ivy/functional/backends/paddle/elementwise.py

```diff
@@ -254,170 +197,94 @@ def sign(
-            return paddle_backend.divide(
-                paddle_backend.log(x), paddle_backend.log(base)
-            ).astype(x.dtype)
-        return paddle.log10(x.astype("float32")).astype(x.dtype)
+    if paddle.is_complex(x):
+        base = paddle.to_tensor(10.0).squeeze()
+        return paddle_backend.divide(
+            paddle_backend.log(x), paddle_backend.log(base)
+        ).astype(x.dtype)
     return paddle.log10(x)
 
 
+@with_supported_dtypes(
+    {"2.5.1 and below": ("float", "complex")},
+    backend_version,
+)
 def log2(x: paddle.Tensor, /, *, out: Optional[paddle.Tensor] = None) -> paddle.Tensor:
-    if x.dtype in [
-        paddle.int8,
-        paddle.int16,
-        paddle.int32,
-        paddle.int64,
-        paddle.uint8,
-        paddle.float16,
-        paddle.complex64,
-        paddle.complex128,
-        paddle.bool,
-    ]:
-        if paddle.is_complex(x):
-            base = paddle.to_tensor(2.0).squeeze()
-            return paddle_backend.divide(
-                paddle_backend.log(x), paddle_backend.log(base)
-            ).astype(x.dtype)
-        return paddle.log2(x.astype("float32")).astype(x.dtype)
+    if paddle.is_complex(x):
+        base = paddle.to_tensor(2.0).squeeze()
+        return paddle_backend.divide(
+            paddle_backend.log(x), paddle_backend.log(base)
+        ).astype(x.dtype)
     return paddle.log2(x)
 
 
+@with_supported_dtypes(
+    {"2.5.1 and below": ("float", "complex")},
+    backend_version,
+)
 def log1p(x: paddle.Tensor, /, *, out: Optional[paddle.Tensor] = None) -> paddle.Tensor:
-    if x.dtype in [
-        paddle.int8,
-        paddle.int16,
-        paddle.int32,
-        paddle.int64,
-        paddle.uint8,
-        paddle.float16,
-        paddle.complex64,
-        paddle.complex128,
-        paddle.bool,
-    ]:
-        if paddle.is_complex(x):
-            return paddle_backend.log(x + 1)
-        return paddle.log1p(x.astype("float32")).astype(x.dtype)
+    if paddle.is_complex(x):
+        return paddle.complex(paddle.log1p(paddle.abs(x)), paddle.angle(x + 1))
     return paddle.log1p(x)
 
 
+@with_supported_dtypes(
+    {
+        "2.5.1 and below": (
+            "float",
+            "int32",
+            "int64",
+            "complex",
+        )
+    },
+    backend_version,
+)
 def isnan(x: paddle.Tensor, /, *, out: Optional[paddle.Tensor] = None) -> paddle.Tensor:
-    if x.dtype in [
-        paddle.int8,
-        paddle.int16,
-        paddle.uint8,
-        paddle.complex64,
-        paddle.complex128,
-        paddle.bool,
-    ]:
-        if paddle.is_complex(x):
-            return paddle.logical_or(paddle.isnan(x.real()), paddle.isnan(x.imag()))
-        return paddle.isnan(x.astype("float32"))
+    if paddle.is_complex(x):
+        return paddle.logical_or(paddle.isnan(x.real()), paddle.isnan(x.imag()))
     return paddle.isnan(x)
 
 
+@with_unsupported_dtypes(
+    {
+        "2.5.1 and below": (
+            "int16",
```

Comment:
have you tested with `int16` if it's unsupported?
Also, `uint32` and `uint64` don't need to be explicitly specified as [invalid](https://github.com/unifyai/ivy/blob/57a3a50423771c561fa1fbbec7077155a6b4dd2e/ivy/functional/backends/paddle/__init__.py#L141) for the entire backend

Reply:
yeah int16 is supported actually